File: doc-examples/example-protobuf-java/src/main/java/example/Coordinates.java
package example;

import io.micronaut.serde.annotation.Serdeable;
import io.micronaut.serde.protobuf.annotation.ProtoField;
import io.micronaut.serde.protobuf.annotation.ProtoType;

// tag::clazz[]
@Serdeable
public record Coordinates(
    @ProtoField(value = 1, type = ProtoType.SFIXED32) int latitudeMicros,   // <1>
    @ProtoField(value = 2, type = ProtoType.SFIXED32) int longitudeMicros
) {
}
// end::clazz[]

File: doc-examples/example-protobuf-kotlin/src/main/kotlin/example/Coordinates.kt
package example

import io.micronaut.serde.annotation.Serdeable
import io.micronaut.serde.protobuf.annotation.ProtoField
import io.micronaut.serde.protobuf.annotation.ProtoType

// tag::clazz[]
@Serdeable
data class Coordinates(
    @field:ProtoField(value = 1, type = ProtoType.SFIXED32) val latitudeMicros: Int,   // <1>
    @field:ProtoField(value = 2, type = ProtoType.SFIXED32) val longitudeMicros: Int
)
// end::clazz[]
